test/test_distributions.py

```diff
@@ -195,6 +195,55 @@ def test_gamma_sample_grad(self):
                                        'rel error {}'.format(rel_error),
                                        'max error {}'.format(rel_error.max())]))
 
+    def test_valid_parameter_broadcasting(self):
+        # Test correct broadcasting of parameter sizes for distributions that have multiple
+        # parameters.
+        # example type (distribution instance, expected sample shape)
+        valid_examples = [
+            (Normal(mean=torch.Tensor([0, 0]), std=torch.Tensor([1])),
+             (2,)),
+            (Normal(mean=torch.Tensor([0, 0]), std=torch.Tensor([[1], [1]])),
+             (2, 2)),
+            (Normal(mean=torch.Tensor([0, 0]), std=torch.Tensor([[1]])),
+             (1, 2)),
+            (Normal(mean=torch.Tensor([0]), std=torch.Tensor([[1]])),
+             (1, 1)),
+            (Gamma(alpha=torch.Tensor([0, 0]), beta=torch.Tensor([[1], [1], [1]])),
```

Comment:
nit: Gamma parameters must be positive. We don't currently check that, but it would be nice to make this test resilient to change in case we do add a check.

Reply:
Will change. Thanks for pointing out.